garak/langproviders/base.py

```diff
@@ -155,6 +243,16 @@ def _load_langprovider(self):
     def _translate(self, text: str) -> str:
         raise NotImplementedError
 
+    def _translate_with_cache(self, text: str) -> str:
+        """Translate text with caching support."""
+        cached_translation = self.cache.get(text)
+        if cached_translation is not None:
+            logging.debug(f"Using cached translation for text: {text[:50]}...")
+            return cached_translation
+        translation = self._translate_impl(text)
+        self.cache.set(text, translation)
```

Comment:
Can we move to a direct src->dst cache? i.e.

```suggestion
        self.cache.[text] = translation
```